src/Sylius/Bundle/ChannelBundle/Behat/ChannelContext.php

```diff
@@ -144,15 +145,35 @@ public function theDefaultChannelHasFollowingConfiguration(TableNode $table)
     }
 
     /**
+     * @Given /^there is a disabled channel "([^""]*)"$/
+     */
+    public function thereIsDisabledChannel($name)
+    {
+        $this->thereIsChannel($name, $name, null, false);
+    }
+
+    /**
+     * @Given /^there is an enabled channel "([^""]*)"$/
+     */
+    public function thereIsEnabledChannel($name)
+    {
+        $this->thereIsChannel($name, $name, null, true);
+    }
+
+    /**
      * @Given /^There is channel "([^""]*)" named "([^""]*)" for url "([^""]*)"$/
      */
-    public function thereIsChannel($code, $name, $url, $locales = null, $currencies = "EUR", $shippingMethods = null, $paymentMethos = null, $flush = true)
+    public function thereIsChannel($code, $name, $url, $enabled = true, $locales = null, $currencies = "EUR", $shippingMethods = null, $paymentMethos = null, $flush = true)
     {
         /* @var $channel ChannelInterface */
-        $channel = $this->getRepository('channel')->createNew();
-        $channel->setCode($code);
-        $channel->setName($name);
-        $channel->setUrl($url);
+        if (null === $channel = $this->getRepository('channel')->findOneBy(array('code' => $code))) {
```

Comment:
If that method will return any other value than `ChannelInterface` or `null` that means it's wrong method! The interface should allow only for one return type (like in every strong typehinted language).

The most readable and the only correct check is:
```php
if (0 === $value)
```